### ISSUE TITLE
Unexpected "unable to reach MRAN" error

### ISSUE THREAD

**Issue by revodavid** [RevolutionAnalytics/checkpoint#188]
In RRO 3.2.1, I get an unexpected "unable to reach MRAN" when using an invalid checkpoint date "20150731" (it should have dashes).

Using the correct `checkpoint("2015-07-31")` incantation eliminates the "unable to find snapshot" error but the "unable to read MRAN" error.
```
> library(checkpoint)

checkpoint: Part of the Reproducible R Toolkit from Revolution Analytics
http://projects.revolutionanalytics.com/rrt/
> checkpoint("20150731")
Can I create directory ~/.checkpoint for internal checkpoint use?(y/n)
y
Scanning for packages used in this project
- Discovered 0 packages
No packages found to install
checkpoint process complete
---
Warning messages:
1: In value[[3L]](cond) : Unable to reach MRAN: cannot open the connection
2: In value[[3L]](cond) : Unable to find snapshot on MRAN
```

**Comment by revodavid** [comment]
```
> sessionInfo()
R version 3.2.1 (2015-06-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 8 x64 (build 9200)

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] viridis_0.1         streamgraph_0.7     htmltools_0.2.6     htmlwidgets_0.5     devtools_1.8.0     
[6] dplyr_0.4.2         checkpoint_0.3.10   RevoUtilsMath_3.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      rstudioapi_0.3.1 xml2_0.1.1       magrittr_1.5     lattice_0.20-33  R6_2.1.0        
 [7] stringr_1.0.0    httr_1.0.0       tools_3.2.1      xts_0.9-7        parallel_3.2.1   grid_3.2.1      
[13] DBI_0.3.1        git2r_0.10.1     rversions_1.0.2  yaml_2.1.13      lazyeval_0.1.10  assertthat_0.1  
[19] digest_0.6.8     tidyr_0.2.0      curl_0.9.1       memoise_0.2.1    stringi_0.5-5    jsonlite_0.9.16 
[25] zoo_1.7-12      
```

**Comment by andrie** [comment]
This happens because MRAN urls with `http://` automatically redirects to `https://`.  However, the mran root url at "http://mran.revolutionanalytics.com/snapshot/" doesn't seem to redirect.

To ilustrate:

```r
library(checkpoint)
options(download.file.method = "wininet", url.method = "wininet")


# Using http --------------------------------------------------------------

# Fails
mran <- "http://mran.revolutionanalytics.com/snapshot/" 
mran.root <- url(mran, method = "wininet")
readLines(mran.root)

# Passes
mran <- "http://mran.revolutionanalytics.com/snapshot/2015-01-01" 
mran.root <- url(mran, method = "wininet")
readLines(mran.root)


# Using https -------------------------------------------------------------

# Fails
mran <- "https://mran.revolutionanalytics.com/snapshot/" 
mran.root <- url(mran, method = "wininet")
readLines(mran.root)

# Passes
mran <- "https://mran.revolutionanalytics.com/snapshot/2015-01-01" 
mran.root <- url(mran, method = "wininet")
readLines(mran.root)

```

**Comment by j-martens** [comment]
Possibly also related to:

https://groups.google.com/forum/#!searchin/rropen/Unable$20to$20reach$20MRAN/rropen/cB0mEpigYyI/7-s9pVrgBQAJ
Later comments on this thread: RevolutionAnalytics/checkpoint#171

**Comment by andrie** [comment]
Fixed in version 0.3.11
